provisioning/device/src/Models/DeviceRegistrationHttp.cs

```diff
@@ -1,35 +1,33 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using Newtonsoft.Json;
+﻿using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
-namespace Microsoft.Azure.Devices.Provisioning.Client.Transport.Models
+namespace Microsoft.Azure.Devices.Provisioning.Client
 {
     /// <summary>
-    /// Device registration.
+    /// Device registration over HTTP.
     /// </summary>
-    internal partial class DeviceRegistration
+    internal class DeviceRegistrationHttp : DeviceRegistration
     {
         /// <summary>
         /// Initializes a new instance of the DeviceRegistration class.
         /// </summary>
-        public DeviceRegistration(string registrationId = default(string), TpmAttestation tpm = default(TpmAttestation), JRaw payload = default(JRaw))
-            : this(payload)
+        public DeviceRegistrationHttp(JRaw payload = default, string registrationId = default, TpmAttestation tpm = default)
+            : base(payload)
         {
             RegistrationId = registrationId;
             Tpm = tpm;
-            CustomInit();
```

Comment:
Some fancy stuff used to be going on to account for partial classes across projects/packages. We don't need to do that anymore. I don't see how `CustomInit()` was even being used, so I removed that too.

Now we rely on class inheritance for these transport-specific classes.